src/fidesops/schemas/dataset.py

```diff
@@ -131,6 +133,19 @@ def valid_data_categories(
         """Validate that all annotated data categories exist in the taxonomy"""
         return _valid_data_categories(v)
 
+    @validator("fidesops_meta")
+    def valid_meta(cls, meta_values: Optional[FidesopsMeta]) -> Optional[FidesopsMeta]:
+        """Validate upfront that the return_all_elements flag can only be specified on array fields"""
+        if not meta_values:
+            return meta_values
+
+        is_array: bool = bool(meta_values.data_type and "[]" in meta_values.data_type)
```

Comment:
small nit: I think `meta_values.data_type.endswith("[]")` might be slightly more performant since it will start at the end of the string and not the start

Reply:
will change!